utilities/geopotential_temp.F90

```diff
@@ -0,0 +1,207 @@
+      integer                      :: cidx            ! CCPP constituent index
+      integer                      :: lyr_step        ! increment to move up a level
+      integer                      :: int_step        ! increment to move up a level
+      real(kind_phys)              :: hkk(ncol)       ! diagonal element of hydrostatic matrix
+      real(kind_phys)              :: hkl(ncol)       ! off-diagonal element
+      real(kind_phys)              :: rog(ncol,pver)  ! Rair / gravit
+      real(kind_phys)              :: tv              ! virtual temperature
+      real(kind_phys)              :: tvfac           ! Tv / T
+      real(kind_phys)              :: qfac(ncol,pver) ! factor to convert from wet to dry mixing ratio
+      real(kind_phys)              :: sum_dry_mixing_ratio(ncol,pver) ! sum of dry water mixing ratios
+      !
+      !-----------------------------------------------------------------------
+      !
+
+      errmsg = ''
+      errflg = 0
+
+      rog(:ncol,:) = rair(:ncol,:) / gravit
+
+      ! Determine order of vertical dimension loop
+      if (layer_surf > layer_toa) then
+         lyr_step = -1
+      else
+         lyr_step = 1
+      end if
+      if (interface_surf > interface_toa) then
+         int_step = -1
+      else
+         int_step = 1
+      end if
+
+      ! The surface height is zero by definition.
+      zi(:ncol, interface_surf) = 0.0_kind_phys
+
+      ! For the computation of generalized virtual temperature (equation 16
+      ! in Lauritzen et al. (2018);  https://doi.org/10.1029/2017MS001257)
+      !
+      ! Compute factor for converting wet to dry mixing ratio (eq.7)
+      !
+      qfac = 1._kind_phys
+      active_flag = .false.
+      do cidx = 1,ncnst
+         !Check if constituent is thermodynamically active:
+         call cprops(cidx)%is_thermo_active(active_flag)
+         if (active_flag) then
+            do klyr = layer_surf, layer_toa, lyr_step
+               do icol = 1, ncol
+                  !Add thermodynamically active species to mixing ratio factor:
+                  qfac(icol, klyr) = qfac(icol, klyr) - carr(icol, klyr, cidx)
+               end do
+            end do
+         end if
+      end do
+      qfac = 1._kind_phys/qfac
+
+      ! Compute sum of thermodynamically active constituent
+      ! mixing ratios with respect to dry air:
+      sum_dry_mixing_ratio = 1._kind_phys
+      active_flag = .false.
+      do cidx = 1,ncnst
+         !Check if constituent is thermodynamically active:
+         call cprops(cidx)%is_thermo_active(active_flag)
+         if (active_flag) then
+            do klyr = layer_surf, layer_toa, lyr_step
+               do icol = 1, ncol
+                  sum_dry_mixing_ratio(icol, klyr) =                          &
+                  sum_dry_mixing_ratio(icol, klyr) +                          &
+                  carr(icol, klyr, cidx) * qfac(icol,klyr)
```

Comment:
I suggest you indent the continuation lines a few spaces so it is more readily apparent that the assignment carries over for a few lines.  You don't need to indent all the way to the = sign, but a few spaces would better offset this assignment.

Reply:
Agreed, normally I try to do that myself, but for whatever reason didn't here.  It should be indented now.